src/cdk/drag-drop/directives/drop-list.ts

```diff
@@ -0,0 +1,309 @@
+    }
+  }
+
+  /** Starts dragging an item. */
+  start(): void {
+    this._dropListRef.start();
+  }
+
+  /**
+   * Drops an item into this container.
+   * @param item Item being dropped into the container.
+   * @param currentIndex Index at which the item should be inserted.
+   * @param previousContainer Container from which the item got dragged in.
+   * @param isPointerOverContainer Whether the user's pointer was over the
+   *    container when the item was dropped.
+   */
+  drop(item: CdkDrag, currentIndex: number, previousContainer: Partial<CdkDropListContainer>,
+    isPointerOverContainer: boolean): void {
+    this._dropListRef.drop(item._dragRef, currentIndex,
+      (previousContainer as CdkDropList)._dropListRef, isPointerOverContainer);
+  }
+
+  /**
+   * Emits an event to indicate that the user moved an item into the container.
+   * @param item Item that was moved into the container.
+   * @param pointerX Position of the item along the X axis.
+   * @param pointerY Position of the item along the Y axis.
+   */
+  enter(item: CdkDrag, pointerX: number, pointerY: number): void {
+    this._dropListRef.enter(item._dragRef, pointerX, pointerY);
+  }
+
+  /**
+   * Removes an item from the container after it was dragged into another container by the user.
+   * @param item Item that was dragged out.
+   */
+  exit(item: CdkDrag): void {
+    this._dropListRef.exit(item._dragRef);
+  }
+
+  /**
+   * Figures out the index of an item in the container.
+   * @param item Item whose index should be determined.
+   */
+  getItemIndex(item: CdkDrag): number {
+    return this._dropListRef.getItemIndex(item._dragRef);
+  }
+
+  /**
+   * Sorts an item inside the container based on its position.
+   * @param item Item to be sorted.
+   * @param pointerX Position of the item along the X axis.
+   * @param pointerY Position of the item along the Y axis.
+   * @param pointerDeta Direction in which the pointer is moving along each axis.
+   */
+  _sortItem(item: CdkDrag, pointerX: number, pointerY: number,
+            pointerDelta: {x: number, y: number}): void {
+    return this._dropListRef._sortItem(item._dragRef, pointerX, pointerY, pointerDelta);
+  }
+
+  /**
+   * Figures out whether an item should be moved into a sibling
+   * drop container, based on its current position.
+   * @param item Drag item that is being moved.
+   * @param x Position of the item along the X axis.
+   * @param y Position of the item along the Y axis.
+   */
+  _getSiblingContainerFromPosition(item: CdkDrag, x: number, y: number):
+    CdkDropListContainer | null {
+    const result = this._dropListRef._getSiblingContainerFromPosition(item._dragRef, x, y);
+    return result ? result.data : null;
+  }
+
+  /**
+   * Checks whether the user's pointer is positioned over the container.
+   * @param x Pointer position along the X axis.
+   * @param y Pointer position along the Y axis.
+   */
+  _isOverContainer(x: number, y: number): boolean {
```

Comment:
Could this function be eliminated?

Reply:
It can be, I left it in to keep the `CdkDropList` compatible with the `CdkDropListContainer` interfaces.